test/widgets/action_sheet_test.dart

```diff
@@ -99,46 +107,244 @@ void main() {
     connection.prepare(httpStatus: 400, json: fakeResponseJson);
   }
 
-  group('AddThumbsUpButton', () {
-    Future<void> tapButton(WidgetTester tester) async {
-      await tester.ensureVisible(find.byIcon(ZulipIcons.smile, skipOffstage: false));
-      await tester.tap(find.byIcon(ZulipIcons.smile));
-      await tester.pump(); // [MenuItemButton.onPressed] called in a post-frame callback: flutter/flutter@e4a39fa2e
-    }
+  group('ReactionButtons', () {
+    final popularCandidates = EmojiStore.popularEmojiCandidates;
 
-    testWidgets('success', (tester) async {
-      final message = eg.streamMessage();
-      await setupToMessageActionSheet(tester, message: message, narrow: TopicNarrow.ofMessage(message));
+    group('popular emoji reactions;', () {
+      testWidgets('ensure all are shown', (tester) async {
+        final message = eg.streamMessage();
+        await setupToMessageActionSheet(tester, message: message, narrow: TopicNarrow.ofMessage(message));
 
-      connection.prepare(json: {});
-      await tapButton(tester);
-      await tester.pump(Duration.zero);
+        // Ensure all popular emoji buttons are shown.
```

Comment:
This is something we already check with the tests below, right? And they do so in a nice end-to-end way: they check that the actual emoji is shown (with an appropriate `Text` widget), and that tapping it works. So I don't think we need an extra test checking that.

One thing I guess this test case does is check that we don't have any *other* options offered here in the same way. But I'd be happy not having a test for that.